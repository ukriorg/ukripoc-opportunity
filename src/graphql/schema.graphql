
type Opportunity @model {
  id: ID!
  name: String!
  description: String
  funders: [Funder] @connection(name: "Funders")
  teammembers: [TeamMember] @connection(name: "TeamMembers")
  type: OpportunityType @connection(name: "OpportunityTypes")
}

type Funder @model {
  id: ID!
  name: string!
}

type TeamMember @model {
  id: ID!
  name: string!
  role: string
}

type OpportunityType @model {
  id: ID!
  name: string!
  description: string!
}