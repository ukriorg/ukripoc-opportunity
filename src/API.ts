/* tslint:disable */
//  This file was automatically generated and should not be edited.

export type CreateOpportunityInput = {
  id?: string | null,
  name: string,
  description?: string | null,
  opportunityTypeId?: string | null,
};

export type UpdateOpportunityInput = {
  id: string,
  name?: string | null,
  description?: string | null,
  opportunityTypeId?: string | null,
};

export type DeleteOpportunityInput = {
  id?: string | null,
};

export type CreateFunderInput = {
  id?: string | null,
  name: string,
  funderOpportunitiesId?: string | null,
};

export type UpdateFunderInput = {
  id: string,
  name?: string | null,
  funderOpportunitiesId?: string | null,
};

export type DeleteFunderInput = {
  id?: string | null,
};

export type CreateTeamMemberInput = {
  id?: string | null,
  name: string,
  role?: string | null,
  teamMemberOpportunityId?: string | null,
};

export type UpdateTeamMemberInput = {
  id: string,
  name?: string | null,
  role?: string | null,
  teamMemberOpportunityId?: string | null,
};

export type DeleteTeamMemberInput = {
  id?: string | null,
};

export type CreateOpportunityTypeInput = {
  id?: string | null,
  name: string,
  description: string,
  opportunityTypeOpportunitiesId?: string | null,
};

export type UpdateOpportunityTypeInput = {
  id: string,
  name?: string | null,
  description?: string | null,
  opportunityTypeOpportunitiesId?: string | null,
};

export type DeleteOpportunityTypeInput = {
  id?: string | null,
};

export type ModelOpportunityFilterInput = {
  id?: ModelIDFilterInput | null,
  name?: ModelStringFilterInput | null,
  description?: ModelStringFilterInput | null,
  and?: Array< ModelOpportunityFilterInput | null > | null,
  or?: Array< ModelOpportunityFilterInput | null > | null,
  not?: ModelOpportunityFilterInput | null,
};

export type ModelIDFilterInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
};

export type ModelStringFilterInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
};

export type ModelFunderFilterInput = {
  id?: ModelIDFilterInput | null,
  name?: ModelStringFilterInput | null,
  and?: Array< ModelFunderFilterInput | null > | null,
  or?: Array< ModelFunderFilterInput | null > | null,
  not?: ModelFunderFilterInput | null,
};

export type ModelTeamMemberFilterInput = {
  id?: ModelIDFilterInput | null,
  name?: ModelStringFilterInput | null,
  role?: ModelStringFilterInput | null,
  and?: Array< ModelTeamMemberFilterInput | null > | null,
  or?: Array< ModelTeamMemberFilterInput | null > | null,
  not?: ModelTeamMemberFilterInput | null,
};

export type ModelOpportunityTypeFilterInput = {
  id?: ModelIDFilterInput | null,
  name?: ModelStringFilterInput | null,
  description?: ModelStringFilterInput | null,
  and?: Array< ModelOpportunityTypeFilterInput | null > | null,
  or?: Array< ModelOpportunityTypeFilterInput | null > | null,
  not?: ModelOpportunityTypeFilterInput | null,
};

export type CreateOpportunityMutationVariables = {
  input: CreateOpportunityInput,
};

export type CreateOpportunityMutation = {
  createOpportunity:  {
    __typename: "Opportunity",
    id: string,
    name: string,
    description: string | null,
    funders:  {
      __typename: "ModelFunderConnection",
      items:  Array< {
        __typename: "Funder",
        id: string,
        name: string,
      } | null > | null,
      nextToken: string | null,
    } | null,
    teammembers:  {
      __typename: "ModelTeamMemberConnection",
      items:  Array< {
        __typename: "TeamMember",
        id: string,
        name: string,
        role: string | null,
      } | null > | null,
      nextToken: string | null,
    } | null,
    type:  {
      __typename: "OpportunityType",
      id: string,
      name: string,
      description: string,
      opportunities:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null,
  } | null,
};

export type UpdateOpportunityMutationVariables = {
  input: UpdateOpportunityInput,
};

export type UpdateOpportunityMutation = {
  updateOpportunity:  {
    __typename: "Opportunity",
    id: string,
    name: string,
    description: string | null,
    funders:  {
      __typename: "ModelFunderConnection",
      items:  Array< {
        __typename: "Funder",
        id: string,
        name: string,
      } | null > | null,
      nextToken: string | null,
    } | null,
    teammembers:  {
      __typename: "ModelTeamMemberConnection",
      items:  Array< {
        __typename: "TeamMember",
        id: string,
        name: string,
        role: string | null,
      } | null > | null,
      nextToken: string | null,
    } | null,
    type:  {
      __typename: "OpportunityType",
      id: string,
      name: string,
      description: string,
      opportunities:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null,
  } | null,
};

export type DeleteOpportunityMutationVariables = {
  input: DeleteOpportunityInput,
};

export type DeleteOpportunityMutation = {
  deleteOpportunity:  {
    __typename: "Opportunity",
    id: string,
    name: string,
    description: string | null,
    funders:  {
      __typename: "ModelFunderConnection",
      items:  Array< {
        __typename: "Funder",
        id: string,
        name: string,
      } | null > | null,
      nextToken: string | null,
    } | null,
    teammembers:  {
      __typename: "ModelTeamMemberConnection",
      items:  Array< {
        __typename: "TeamMember",
        id: string,
        name: string,
        role: string | null,
      } | null > | null,
      nextToken: string | null,
    } | null,
    type:  {
      __typename: "OpportunityType",
      id: string,
      name: string,
      description: string,
      opportunities:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null,
  } | null,
};

export type CreateFunderMutationVariables = {
  input: CreateFunderInput,
};

export type CreateFunderMutation = {
  createFunder:  {
    __typename: "Funder",
    id: string,
    name: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type UpdateFunderMutationVariables = {
  input: UpdateFunderInput,
};

export type UpdateFunderMutation = {
  updateFunder:  {
    __typename: "Funder",
    id: string,
    name: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type DeleteFunderMutationVariables = {
  input: DeleteFunderInput,
};

export type DeleteFunderMutation = {
  deleteFunder:  {
    __typename: "Funder",
    id: string,
    name: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type CreateTeamMemberMutationVariables = {
  input: CreateTeamMemberInput,
};

export type CreateTeamMemberMutation = {
  createTeamMember:  {
    __typename: "TeamMember",
    id: string,
    name: string,
    role: string | null,
    opportunity:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type UpdateTeamMemberMutationVariables = {
  input: UpdateTeamMemberInput,
};

export type UpdateTeamMemberMutation = {
  updateTeamMember:  {
    __typename: "TeamMember",
    id: string,
    name: string,
    role: string | null,
    opportunity:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type DeleteTeamMemberMutationVariables = {
  input: DeleteTeamMemberInput,
};

export type DeleteTeamMemberMutation = {
  deleteTeamMember:  {
    __typename: "TeamMember",
    id: string,
    name: string,
    role: string | null,
    opportunity:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type CreateOpportunityTypeMutationVariables = {
  input: CreateOpportunityTypeInput,
};

export type CreateOpportunityTypeMutation = {
  createOpportunityType:  {
    __typename: "OpportunityType",
    id: string,
    name: string,
    description: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type UpdateOpportunityTypeMutationVariables = {
  input: UpdateOpportunityTypeInput,
};

export type UpdateOpportunityTypeMutation = {
  updateOpportunityType:  {
    __typename: "OpportunityType",
    id: string,
    name: string,
    description: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type DeleteOpportunityTypeMutationVariables = {
  input: DeleteOpportunityTypeInput,
};

export type DeleteOpportunityTypeMutation = {
  deleteOpportunityType:  {
    __typename: "OpportunityType",
    id: string,
    name: string,
    description: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type GetOpportunityQueryVariables = {
  id: string,
};

export type GetOpportunityQuery = {
  getOpportunity:  {
    __typename: "Opportunity",
    id: string,
    name: string,
    description: string | null,
    funders:  {
      __typename: "ModelFunderConnection",
      items:  Array< {
        __typename: "Funder",
        id: string,
        name: string,
      } | null > | null,
      nextToken: string | null,
    } | null,
    teammembers:  {
      __typename: "ModelTeamMemberConnection",
      items:  Array< {
        __typename: "TeamMember",
        id: string,
        name: string,
        role: string | null,
      } | null > | null,
      nextToken: string | null,
    } | null,
    type:  {
      __typename: "OpportunityType",
      id: string,
      name: string,
      description: string,
      opportunities:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null,
  } | null,
};

export type ListOpportunitysQueryVariables = {
  filter?: ModelOpportunityFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListOpportunitysQuery = {
  listOpportunitys:  {
    __typename: "ModelOpportunityConnection",
    items:  Array< {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null > | null,
    nextToken: string | null,
  } | null,
};

export type GetFunderQueryVariables = {
  id: string,
};

export type GetFunderQuery = {
  getFunder:  {
    __typename: "Funder",
    id: string,
    name: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type ListFundersQueryVariables = {
  filter?: ModelFunderFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListFundersQuery = {
  listFunders:  {
    __typename: "ModelFunderConnection",
    items:  Array< {
      __typename: "Funder",
      id: string,
      name: string,
      opportunities:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null > | null,
    nextToken: string | null,
  } | null,
};

export type GetTeamMemberQueryVariables = {
  id: string,
};

export type GetTeamMemberQuery = {
  getTeamMember:  {
    __typename: "TeamMember",
    id: string,
    name: string,
    role: string | null,
    opportunity:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type ListTeamMembersQueryVariables = {
  filter?: ModelTeamMemberFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListTeamMembersQuery = {
  listTeamMembers:  {
    __typename: "ModelTeamMemberConnection",
    items:  Array< {
      __typename: "TeamMember",
      id: string,
      name: string,
      role: string | null,
      opportunity:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null > | null,
    nextToken: string | null,
  } | null,
};

export type GetOpportunityTypeQueryVariables = {
  id: string,
};

export type GetOpportunityTypeQuery = {
  getOpportunityType:  {
    __typename: "OpportunityType",
    id: string,
    name: string,
    description: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type ListOpportunityTypesQueryVariables = {
  filter?: ModelOpportunityTypeFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListOpportunityTypesQuery = {
  listOpportunityTypes:  {
    __typename: "ModelOpportunityTypeConnection",
    items:  Array< {
      __typename: "OpportunityType",
      id: string,
      name: string,
      description: string,
      opportunities:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null > | null,
    nextToken: string | null,
  } | null,
};

export type OnCreateOpportunitySubscription = {
  onCreateOpportunity:  {
    __typename: "Opportunity",
    id: string,
    name: string,
    description: string | null,
    funders:  {
      __typename: "ModelFunderConnection",
      items:  Array< {
        __typename: "Funder",
        id: string,
        name: string,
      } | null > | null,
      nextToken: string | null,
    } | null,
    teammembers:  {
      __typename: "ModelTeamMemberConnection",
      items:  Array< {
        __typename: "TeamMember",
        id: string,
        name: string,
        role: string | null,
      } | null > | null,
      nextToken: string | null,
    } | null,
    type:  {
      __typename: "OpportunityType",
      id: string,
      name: string,
      description: string,
      opportunities:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null,
  } | null,
};

export type OnUpdateOpportunitySubscription = {
  onUpdateOpportunity:  {
    __typename: "Opportunity",
    id: string,
    name: string,
    description: string | null,
    funders:  {
      __typename: "ModelFunderConnection",
      items:  Array< {
        __typename: "Funder",
        id: string,
        name: string,
      } | null > | null,
      nextToken: string | null,
    } | null,
    teammembers:  {
      __typename: "ModelTeamMemberConnection",
      items:  Array< {
        __typename: "TeamMember",
        id: string,
        name: string,
        role: string | null,
      } | null > | null,
      nextToken: string | null,
    } | null,
    type:  {
      __typename: "OpportunityType",
      id: string,
      name: string,
      description: string,
      opportunities:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null,
  } | null,
};

export type OnDeleteOpportunitySubscription = {
  onDeleteOpportunity:  {
    __typename: "Opportunity",
    id: string,
    name: string,
    description: string | null,
    funders:  {
      __typename: "ModelFunderConnection",
      items:  Array< {
        __typename: "Funder",
        id: string,
        name: string,
      } | null > | null,
      nextToken: string | null,
    } | null,
    teammembers:  {
      __typename: "ModelTeamMemberConnection",
      items:  Array< {
        __typename: "TeamMember",
        id: string,
        name: string,
        role: string | null,
      } | null > | null,
      nextToken: string | null,
    } | null,
    type:  {
      __typename: "OpportunityType",
      id: string,
      name: string,
      description: string,
      opportunities:  {
        __typename: "Opportunity",
        id: string,
        name: string,
        description: string | null,
      } | null,
    } | null,
  } | null,
};

export type OnCreateFunderSubscription = {
  onCreateFunder:  {
    __typename: "Funder",
    id: string,
    name: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type OnUpdateFunderSubscription = {
  onUpdateFunder:  {
    __typename: "Funder",
    id: string,
    name: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type OnDeleteFunderSubscription = {
  onDeleteFunder:  {
    __typename: "Funder",
    id: string,
    name: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type OnCreateTeamMemberSubscription = {
  onCreateTeamMember:  {
    __typename: "TeamMember",
    id: string,
    name: string,
    role: string | null,
    opportunity:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type OnUpdateTeamMemberSubscription = {
  onUpdateTeamMember:  {
    __typename: "TeamMember",
    id: string,
    name: string,
    role: string | null,
    opportunity:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type OnDeleteTeamMemberSubscription = {
  onDeleteTeamMember:  {
    __typename: "TeamMember",
    id: string,
    name: string,
    role: string | null,
    opportunity:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type OnCreateOpportunityTypeSubscription = {
  onCreateOpportunityType:  {
    __typename: "OpportunityType",
    id: string,
    name: string,
    description: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type OnUpdateOpportunityTypeSubscription = {
  onUpdateOpportunityType:  {
    __typename: "OpportunityType",
    id: string,
    name: string,
    description: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};

export type OnDeleteOpportunityTypeSubscription = {
  onDeleteOpportunityType:  {
    __typename: "OpportunityType",
    id: string,
    name: string,
    description: string,
    opportunities:  {
      __typename: "Opportunity",
      id: string,
      name: string,
      description: string | null,
      funders:  {
        __typename: "ModelFunderConnection",
        nextToken: string | null,
      } | null,
      teammembers:  {
        __typename: "ModelTeamMemberConnection",
        nextToken: string | null,
      } | null,
      type:  {
        __typename: "OpportunityType",
        id: string,
        name: string,
        description: string,
      } | null,
    } | null,
  } | null,
};
