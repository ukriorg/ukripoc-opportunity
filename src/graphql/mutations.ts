// tslint:disable
// this is an auto generated file. This will be overwritten

export const createOpportunity = `mutation CreateOpportunity($input: CreateOpportunityInput!) {
  createOpportunity(input: $input) {
    id
    name
    description
    funders {
      items {
        id
        name
      }
      nextToken
    }
    teammembers {
      items {
        id
        name
        role
      }
      nextToken
    }
    type {
      id
      name
      description
      opportunities {
        id
        name
        description
      }
    }
  }
}
`;
export const updateOpportunity = `mutation UpdateOpportunity($input: UpdateOpportunityInput!) {
  updateOpportunity(input: $input) {
    id
    name
    description
    funders {
      items {
        id
        name
      }
      nextToken
    }
    teammembers {
      items {
        id
        name
        role
      }
      nextToken
    }
    type {
      id
      name
      description
      opportunities {
        id
        name
        description
      }
    }
  }
}
`;
export const deleteOpportunity = `mutation DeleteOpportunity($input: DeleteOpportunityInput!) {
  deleteOpportunity(input: $input) {
    id
    name
    description
    funders {
      items {
        id
        name
      }
      nextToken
    }
    teammembers {
      items {
        id
        name
        role
      }
      nextToken
    }
    type {
      id
      name
      description
      opportunities {
        id
        name
        description
      }
    }
  }
}
`;
export const createFunder = `mutation CreateFunder($input: CreateFunderInput!) {
  createFunder(input: $input) {
    id
    name
    opportunities {
      id
      name
      description
      funders {
        nextToken
      }
      teammembers {
        nextToken
      }
      type {
        id
        name
        description
      }
    }
  }
}
`;
export const updateFunder = `mutation UpdateFunder($input: UpdateFunderInput!) {
  updateFunder(input: $input) {
    id
    name
    opportunities {
      id
      name
      description
      funders {
        nextToken
      }
      teammembers {
        nextToken
      }
      type {
        id
        name
        description
      }
    }
  }
}
`;
export const deleteFunder = `mutation DeleteFunder($input: DeleteFunderInput!) {
  deleteFunder(input: $input) {
    id
    name
    opportunities {
      id
      name
      description
      funders {
        nextToken
      }
      teammembers {
        nextToken
      }
      type {
        id
        name
        description
      }
    }
  }
}
`;
export const createTeamMember = `mutation CreateTeamMember($input: CreateTeamMemberInput!) {
  createTeamMember(input: $input) {
    id
    name
    role
    opportunity {
      id
      name
      description
      funders {
        nextToken
      }
      teammembers {
        nextToken
      }
      type {
        id
        name
        description
      }
    }
  }
}
`;
export const updateTeamMember = `mutation UpdateTeamMember($input: UpdateTeamMemberInput!) {
  updateTeamMember(input: $input) {
    id
    name
    role
    opportunity {
      id
      name
      description
      funders {
        nextToken
      }
      teammembers {
        nextToken
      }
      type {
        id
        name
        description
      }
    }
  }
}
`;
export const deleteTeamMember = `mutation DeleteTeamMember($input: DeleteTeamMemberInput!) {
  deleteTeamMember(input: $input) {
    id
    name
    role
    opportunity {
      id
      name
      description
      funders {
        nextToken
      }
      teammembers {
        nextToken
      }
      type {
        id
        name
        description
      }
    }
  }
}
`;
export const createOpportunityType = `mutation CreateOpportunityType($input: CreateOpportunityTypeInput!) {
  createOpportunityType(input: $input) {
    id
    name
    description
    opportunities {
      id
      name
      description
      funders {
        nextToken
      }
      teammembers {
        nextToken
      }
      type {
        id
        name
        description
      }
    }
  }
}
`;
export const updateOpportunityType = `mutation UpdateOpportunityType($input: UpdateOpportunityTypeInput!) {
  updateOpportunityType(input: $input) {
    id
    name
    description
    opportunities {
      id
      name
      description
      funders {
        nextToken
      }
      teammembers {
        nextToken
      }
      type {
        id
        name
        description
      }
    }
  }
}
`;
export const deleteOpportunityType = `mutation DeleteOpportunityType($input: DeleteOpportunityTypeInput!) {
  deleteOpportunityType(input: $input) {
    id
    name
    description
    opportunities {
      id
      name
      description
      funders {
        nextToken
      }
      teammembers {
        nextToken
      }
      type {
        id
        name
        description
      }
    }
  }
}
`;
