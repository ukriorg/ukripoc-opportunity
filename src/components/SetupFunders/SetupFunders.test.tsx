import React from "react";
import renderer from "react-test-renderer";
import { SetupFunders } from "./index";
import fundersList from "../../fixtures/funders.json";
import { GetOpportunityQuery } from "../../API";
import { updateOpportunity } from "../../graphql/mutations";

const opportunity: GetOpportunityQuery = {
    getOpportunity: {
        id: "675c0700-09c3-4b1c-9292-71f96ef0567e",
        name: "New opp",
        description: "Today",
        funders: ["Other"],
        fundersComplete: null,
        teammembers: {
            items: [],
            nextToken: null,
            __typename: "ModelTeamMemberConnection"
        },
        teammembersComplete: null,
        type: null,
        typeComplete: null,
        __typename: "Opportunity"
    }
};

describe("SetupFunders", () =>
    it("renders correctly", () => {
        const tree = renderer
            .create(
                <SetupFunders
                    funders={fundersList}
                    currentOpportunity={opportunity}
                    fundersChanged={() => {}}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    }));
