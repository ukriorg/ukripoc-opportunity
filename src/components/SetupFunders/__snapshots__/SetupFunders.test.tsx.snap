// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupFunders renders correctly 1`] = `
<div>
  <h1>
    Funders
  </h1>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={false}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      AHRC
    </span>
  </label>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={false}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      BBSRC
    </span>
  </label>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={false}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      EPSRC
    </span>
  </label>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={false}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      Innovate UK
    </span>
  </label>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={false}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      MRC
    </span>
  </label>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={false}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      NERC
    </span>
  </label>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={false}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      Research England
    </span>
  </label>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={false}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      STFC
    </span>
  </label>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={false}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      UKRI
    </span>
  </label>
  <label
    className="sc-bwzfXH kZZPar"
  >
    <input
      checked={true}
      className="sc-htpNat bZWrQc"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-bxivhb ckQlQp"
    >
      Other
    </span>
  </label>
  <button
    className="sc-EHOje hEaLol"
    disabled={false}
    onClick={[Function]}
  >
    Done
  </button>
</div>
`;
